src/crh/poseidon/sbox/constraints.rs

```diff
@@ -5,60 +5,43 @@ use ark_r1cs_std::fields::FieldVar;
 use ark_relations::r1cs::SynthesisError;
 
 pub trait SboxConstraints {
-    fn synthesize_sbox<F: PrimeField>(
-        &self,
-        input: FpVar<F>,
-        round_key: F,
-    ) -> Result<FpVar<F>, SynthesisError>;
+    fn synthesize_sbox<F: PrimeField>(&self, input: FpVar<F>) -> Result<FpVar<F>, SynthesisError>;
 }
 
 impl SboxConstraints for PoseidonSbox {
     fn synthesize_sbox<F: PrimeField>(
         &self,
         input_var: FpVar<F>,
-        round_key: F,
     ) -> Result<FpVar<F>, SynthesisError> {
         match self {
             PoseidonSbox::Exponentiation(val) => match val {
-                3 => synthesize_exp3_sbox::<F>(input_var, round_key),
-                5 => synthesize_exp5_sbox::<F>(input_var, round_key),
-                _ => synthesize_exp3_sbox::<F>(input_var, round_key),
+                3 => synthesize_exp3_sbox::<F>(input_var),
+                5 => synthesize_exp5_sbox::<F>(input_var),
+                _ => synthesize_exp3_sbox::<F>(input_var),
             },
-            PoseidonSbox::Inverse => synthesize_inverse_sbox::<F>(input_var, round_key),
+            PoseidonSbox::Inverse => synthesize_inverse_sbox::<F>(input_var),
         }
     }
 }
 
 // Allocate variables in circuit and enforce constraints when Sbox as cube
-fn synthesize_exp3_sbox<F: PrimeField>(
-    input_var: FpVar<F>,
-    round_key: F,
-) -> Result<FpVar<F>, SynthesisError> {
-    let inp_plus_const: FpVar<F> = input_var + round_key;
-    let sqr = inp_plus_const.clone() * inp_plus_const.clone();
-    let cube = inp_plus_const.clone() * sqr;
+fn synthesize_exp3_sbox<F: PrimeField>(input_var: FpVar<F>) -> Result<FpVar<F>, SynthesisError> {
+    let sqr = input_var.clone() * input_var.clone();
+    let cube = input_var.clone() * sqr;
```

Comment:
I'm a bit confused about why we need so many clones

Reply:
Passing reference now